src/cephadm/cephadm

```diff
@@ -6867,6 +6867,13 @@ def command_rm_daemon(ctx):
     else:
         call_throws(ctx, ['rm', '-rf', data_dir])
 
+    if 'tcp_ports' in ctx and ctx.tcp_ports is not None:
+        ports: List[int] = [int(p) for p in ctx.tcp_ports.split()]
+        fw = Firewalld(ctx)
+        fw.close_ports(ports)
```

Comment:
one question is what do we want to do if this fails?  In `_remove_daemon` in serve.py we only remove the daemon from our daemon cache if we don't get a nonzero return code running `rm-daemon`. Do we want to be throwing an error if we can't close the ports, or just log and continue, given we've already done the rest of the daemon removal at this point.

Reply:
I changed the code. I think it's better to still remove the daemon if we are not able to close the ports (because that was already the legacy behavior). In addition I think that makes sense, because we can end-up with a stalled daemon because we are not able to free some port.